packages/grid/_modules_/grid/hooks/features/selection/useGridSelection.ts

```diff
@@ -190,8 +205,27 @@ export const useGridSelection = (apiRef: GridApiRef): void => {
 
   React.useEffect(() => {
     const currentModel = Object.values(apiRef.current.getState().selection);
-    if (!isDeepEqual(currentModel, options.selectionModel)) {
-      apiRef.current.setSelectionModel(options.selectionModel || []);
+    if (!isDeepEqual(currentModel, selectionModel)) {
+      apiRef.current.setSelectionModel(selectionModel || []);
     }
-  }, [apiRef, options.selectionModel]);
+  }, [apiRef, selectionModel]);
+
+  React.useEffect(() => {
+    setGridState((state) => {
+      const newSelectionState = { ...state.selection };
+      let hasChanged = false;
+      Object.keys(newSelectionState).forEach((id: GridRowId) => {
+        const isSelectable = !isRowSelectable || isRowSelectable(apiRef.current.getRowParams(id));
+        if (!isSelectable) {
+          delete newSelectionState[id];
+          hasChanged = true;
+        }
+      });
+      if (hasChanged) {
+        return { ...state, selection: newSelectionState };
+      }
+      return state;
+    });
+    forceUpdate();
+  }, [apiRef, setGridState, options.isRowSelectable, forceUpdate, isRowSelectable]);
```

Comment:
do we need `options.isRowSelectable` and `isRowSelectable`?

Reply:
I missed this one.